switchapi/switch_tdi.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022-2024 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __SWITCH_TDI_H__
+#define __SWITCH_TDI_H__
+
+// tdi_info.h does not include the header files it depends on,
```

Comment:
Maybe this comment can be removed? The filename reference doesnt seem to be align with the content

Reply:
It explains why I'm doing this:
```c
// clang-format off
```
clang-format alphabetizes the #includes, moving `tdi_defs.h` to the end (after `tdi_info.h`). `tdi_info` uses symbols defined in `tdi_defs.h`, but it doesn't #include the file (which is very bad form). I work around the problem by disabling clang-format.

Reply:
Ah, I misunderstood the comment.

Reply:
Perhaps:
```c
// Disable clang-format so it doesn't alphabetize the #includes. tdi_defs.h must precede the
// other header files because one or more of them depend on tdi_defs.h but don't #include it.
```